sci-libs/gdal/gdal-2.0.2-r1.ebuild

```diff
@@ -0,0 +1,321 @@
+	#     also that thing is developed by the gdal people
+	# kakadu, mrsid jp2mrsid - another jpeg2k stuff, ignore
+	# bsb - legal issues
+	# oracle - disabled, i dont have and can't test
+	# ingres - same story as oracle oci
+	# podofo - we use poppler instead they are exclusive for each other
+	# tiff is a hard dep
+	ECONF_SOURCE="${S}" econf \
+		--includedir="${EPREFIX}/usr/include/${PN}" \
+		--disable-static \
+		--enable-shared \
+		--with-expat \
+		--with-geotiff \
+		--with-grib \
+		--with-libtiff \
+		--with-libz="${EPREFIX}/usr/" \
+		--with-ogr \
+		--without-bsb \
+		--without-dods-root \
+		--without-dwgdirect \
+		--without-epsilon \
+		--without-fme \
+		--without-grass \
+		--without-hdf4 \
+		--without-idb \
+		--without-ingres \
+		--without-jp2mrsid \
+		--without-kakadu \
+		--without-libtool \
+		--without-mrsid \
+		--without-msg \
+		--without-oci \
+		--without-pcraster \
+		--without-podofo \
+		--without-sde \
+		$(use_enable debug) \
+		$(use_with armadillo) \
+		$(use_with aux_xml pam) \
+		$(use_with curl) \
+		--without-ecw \
+		$(use_with fits cfitsio) \
+		$(use_with geos) \
+		$(use_with gif) \
+		$(use_with gml xerces) \
+		$(use_with hdf5) \
+		$(use_with jpeg pcidsk) \
+		$(use_with jpeg) \
+		$(use_with jpeg2k jasper) \
+		$(use_with mysql mysql "${EPREFIX}"/usr/bin/mysql_config) \
+		$(use_with netcdf) \
+		$(use_with odbc) \
+		$(use_with ogdi ogdi "${EPREFIX}"/usr) \
+		$(use_with opencl) \
+		$(use_with pdf poppler) \
+		$(use_with perl) \
+		$(use_with png) \
+		$(use_with postgres pg) \
+		$(use_with spatialite) \
+		$(use_with sqlite sqlite3 "${EPREFIX}"/usr) \
+		$(use_with threads) \
+		$(use_with xls freexl) \
+		${myopts}
+
+	# mysql-config puts this in (and boy is it a PITA to get it out)
+	if use mysql; then
+		sed -i \
+			-e "s: -rdynamic : :" \
+			GDALmake.opt || die "sed LIBS failed"
+	fi
+
+	if [[ -n $use_python ]]; then
+		# updated for newer swig (must specify the path to input files)
+		sed -i \
+			-e "s: gdal_array.i: ../include/gdal_array.i:" \
+			-e "s:\$(DESTDIR)\$(prefix):\$(DESTDIR)\$(INST_PREFIX):g" \
+			swig/python/GNUmakefile || die "sed python makefile failed"
+		sed -i \
+			-e "s:library_dirs = :library_dirs = /usr/$(get_libdir):g" \
+			swig/python/setup.cfg || die "sed python setup.cfg failed"
+#			-e "s:gdal_config=.*$:gdal_config=../../../apps/gdal-config:g" \
+	fi
+}
+
+src_configure() {
+	local use_python=""
+
+	gdal_src_configure
+
+	if use python; then
+		use_python="yes"
+		python_foreach_impl run_in_build_dir gdal_src_configure
+	fi
+}
+
+src_compile() {
+	if use perl; then
+		rm "${S}"/swig/perl/*_wrap.cpp || die
+		emake -C "${S}"/swig/perl generate
+	fi
+
+	default
+
+	if use perl ; then
+		pushd "${S}"/swig/perl > /dev/null
+		perl-module_src_configure
+		perl-module_src_compile
+		popd > /dev/null
+	fi
+
+	use doc && emake docs
+
+	compile_python() {
+		rm -f swig/python/*_wrap.cpp || die
+		emake -C swig/python generate
+		emake -C swig/python build
```

Comment:
The build part do via distutils-r1. Kill the whole foreach thing, and just do sth like:

    if use python; then
        cd "${S}"/swig/perl || die
        distutils-r1_src_compile
    fi

(note: looking at other code, maintainer may prefer pushd/popd, don't forget to ||die)